### PR TITLE
(NOT FOR MERGE) Bindings-Only RL updates for 0.0.103

### PR DESCRIPTION
**Pull request by TheBlueMatt**
The first commit works around a rustc bug, the second a limitation in the bindings, but I don't really have time to fix it pre-0.0.103. Seeking some level of review then I'll push to a separate branch for bindings to build against.

### CODE REVIEW

File: lightning-invoice/src/payment.rs [hunk 1]
```diff
@@ -131,17 +131,14 @@ use std::sync::Mutex;
 use std::time::{Duration, SystemTime};
 
 /// A utility for paying [`Invoice]`s.
-pub struct InvoicePayer<P: Deref, R, S: Deref, L: Deref, E>
+pub struct InvoicePayer<P: Deref, R: Router<S>, S: routing::Score, Sc: Deref<Target=LockableScore<S>>, L: Deref, E: EventHandler>
```
Comment (jkczyz) [inline lightning-invoice/src/payment.rs:134]: Probably could dispense with `LockableScore` entirely and use `Mutex` directly. Would just require an `unwrap` after `lock` in the code.

**Comment by jkczyz** [comment]
Generally looks sane aside from the build breakage.

File: lightning-invoice/src/payment.rs (hunk 1 above)
Comment (TheBlueMatt) [reply lightning-invoice/src/payment.rs:137]: The bindings don't have any logic for exposing a native `Mutex`, so we have to wrap it :/

File: lightning/src/routing/mod.rs [hunk 2]
```diff
@@ -53,3 +56,10 @@ impl<S: Score> LockableScore<S> {
 		self.scorer.lock().unwrap()
 	}
 }
+
+impl<S: Score> Writeable for LockableScore<S> {
```
Comment (valentinewallace) [inline lightning/src/routing/mod.rs:60]: Hmm, this seems like something we might want upstream?

**Comment by valentinewallace** [comment]
Seems legit 50b1017fca4c7b73f9dc71ed923e91c6c1da6459

File: lightning/src/routing/mod.rs (hunk 2 above)
Comment (TheBlueMatt) [reply lightning/src/routing/mod.rs:60]: Yea, I think so, eventually at least. Either way we need it for 0.0.103 bindings, so gotta have it here.

**Comment by TheBlueMatt** [comment]
(note that no-std is broken but I'm calling that totally fine for bindings for now, sadly)

**Comment by TheBlueMatt** [comment]
This now lives at the 0.0.103-bindings branch.

**Pull request closed without merging**